packages/clerk-js/src/ui/UserProfile/MfaBackupCodeCreatePage.tsx

```diff
@@ -0,0 +1,181 @@
+import { BackupCodeResource } from '@clerk/types';
+import React from 'react';
+
+import { PrintableComponent, usePrintable } from '../common';
+import { useCoreUser, useEnvironment } from '../contexts';
+import { Button, Col, Flex, Grid, Heading, localizationKeys, Spinner, Text } from '../customizables';
+import { useCardState } from '../elements';
+import { useClipboard } from '../hooks';
+import { mqu } from '../styledSystem';
+import { getIdentifier, handleError } from '../utils';
+import { FormButtonContainer } from './FormButtons';
+import { NavigateToFlowStartButton } from './NavigateToFlowStartButton';
+import { ContentPage } from './Page';
+
+export const MfaBackupCodeCreatePage = () => {
+  const user = useCoreUser();
+  const card = useCardState();
+  const [backupCode, setBackupCode] = React.useState<BackupCodeResource | undefined>(undefined);
+
+  const title = 'Add multi-factor authentication';
+  const text =
+    'Backup codes are now enabled. When signing in, you will need to enter a backup code as an additional step. Each code can only be used once.';
+
+  React.useEffect(() => {
+    if (backupCode) {
+      return;
+    }
+
+    void user
+      .createBackupCode()
+      .then((backupCode: BackupCodeResource) => setBackupCode(backupCode))
+      .catch(err => handleError(err, [], card.setError));
+  }, []);
+
+  if (card.error) {
+    return <ContentPage.Root headerTitle={title} />;
+  }
+
+  if (!backupCode) {
+    return (
+      <Spinner
+        colorScheme='primary'
+        size='lg'
+      />
+    );
+  }
+
+  return (
+    <ContentPage.Root headerTitle={title}>
+      <Text
+        localizationKey={text}
+        variant='regularRegular'
+      />
+      <MfaBackupCode backupCodes={backupCode.codes} />
+
+      <FormButtonContainer>
+        <NavigateToFlowStartButton
+          variant='solid'
+          autoFocus
+          localizationKey={localizationKeys('userProfile.formButtonPrimary__finish')}
+        />
+      </FormButtonContainer>
+    </ContentPage.Root>
+  );
+};
+
+type MfaBackupCodeProps = {
+  backupCodes?: string[];
+};
+
+const MfaBackupCode = (props: MfaBackupCodeProps) => {
+  const { backupCodes } = props;
+  const { applicationName } = useEnvironment().displayConfig;
+  const user = useCoreUser();
+  const { print, printableProps } = usePrintable();
+  const { onCopy, hasCopied } = useClipboard(backupCodes?.join(',') as string);
```

Comment:
```suggestion
  const { onCopy, hasCopied } = useClipboard(backupCodes?.join(',') || '');
```

Reply:
That way we can get rid of the manual type assertion and be safe if `backupCodes` end up as undefined for some reason.